projects/packages/backup/tests/php/test-rest-controller.php

```diff
@@ -166,6 +172,219 @@ public function test_delete_backup_helper_script_success() {
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/options/backup` endpoint with admin user.
+	 */
+	public function test_backup_options_unauthorized() {
+		wp_set_current_user( $this->admin_id );
+
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/options/backup' );
+		$request->set_query_params(
+			array(
+				'name' => 'home',
+			)
+		);
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 403, $response->get_status() );
+		$this->assertEquals( 'You are not allowed to perform this action.', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/options/backup` endpoint on success.
+	 */
+	public function test_backup_options_success() {
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/options/backup' );
+		$request->set_query_params(
+			array(
+				'name' => 'home',
+			)
+		);
+		$response = $this->server->dispatch( $request );
+
+		$response = $this->dispatch_request_signed_with_blog_token( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/comments/(?P<id>\d+)/backup` endpoint with admin user.
+	 */
+	public function test_backup_comment_unauthorized() {
+		wp_set_current_user( $this->admin_id );
+
+		$request  = new WP_REST_Request( 'GET', '/jetpack/v4/comments/1234/backup' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 403, $response->get_status() );
+		$this->assertEquals( 'You are not allowed to perform this action.', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/comments/(?P<id>\d+)/backup` endpoint on success.
+	 */
+	public function test_backup_comment_success() {
+		$request = new WP_REST_Request( 'GET', '/jetpack/v4/comments/1234/backup' );
+
+		$response = $this->server->dispatch( $request );
+
+		$response = $this->dispatch_request_signed_with_blog_token( $request );
+
+		$this->assertEquals( 404, $response->get_status() ); // success in this context.
+
+		$this->assertEquals( 'Comment not found', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/posts/(?P<id>\d+)/backup` endpoint with admin user.
+	 */
+	public function test_backup_post_unauthorized() {
+		wp_set_current_user( $this->admin_id );
+
+		$request  = new WP_REST_Request( 'GET', '/jetpack/v4/posts/1/backup' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertEquals( 403, $response->get_status() );
+		$this->assertEquals( 'You are not allowed to perform this action.', $response->get_data()['message'] );
+	}
+
+	/**
+	 * Testing the `/jetpack/v4/posts/(?P<id>\d+)/backup endpoint on success.
+	 */
+	public function test_backup_post_success() {
+		$post_id = wp_insert_post(
+			array(
+				'post_content' => 'dummy',
+			)
+		);
+
+		$request = new WP_REST_Request( 'GET', "/jetpack/v4/posts/{$post_id}/backup" );
+
+		$response = $this->server->dispatch( $request );
```

Comment:
I think this `server->dispatch` call is unnecessary, since we call `dispatch_request_signed_with_blog_token` below.